kolibri/core/device/migrations/0021_default_demographic_fields.py

```diff
@@ -0,0 +1,192 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.29 on 2024-03-28 15:35
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+import kolibri.core.auth.constants.demographics
+import kolibri.core.fields
+import kolibri.core.utils.validators
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("device", "0020_fix_learner_device_status_choices"),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name="devicesettings",
+            name="extra_settings",
+            field=kolibri.core.fields.JSONField(
+                default={
+                    "allow_download_on_metered_connection": False,
+                    "allow_learner_download_resources": False,
+                    "enable_automatic_download": True,
+                    "limit_for_autodownload": 0,
+                    "set_limit_for_autodownload": False,
+                },
+                validators=[
+                    kolibri.core.utils.validators.JSON_Schema_Validator(
+                        {
+                            "properties": {
+                                "allow_download_on_metered_connection": {
+                                    "type": "boolean"
+                                },
+                                "allow_learner_download_resources": {
+                                    "optional": True,
+                                    "type": "boolean",
+                                },
+                                "default_demographic_field_schema": {
+                                    "items": {
+                                        "properties": {
+                                            "description": {"type": "string"},
+                                            "enumValues": {
+                                                "items": {
+                                                    "properties": {
+                                                        "defaultLabel": {
+                                                            "type": "string"
+                                                        },
+                                                        "translations": {
+                                                            "items": {
+                                                                "properties": {
+                                                                    "language": {
+                                                                        "enum": [
```

Comment:
Not blocking as it's a reasonably unlikely edge case.

Looks like if we ever add new languages then we'll need to update this list? Do we have any way to be sure we keep this list up-to-date w/ our available languages?

Reply:
If we update the list of languages, Django will complain that we have model changes that need migrations, and we'll have to generate a new migration - this will fail CI checks, so we will notice :)